### PR TITLE
fix: allow openapi version to be overridden

### LINKED ISSUE #601: Cannot set OpenAPI version
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.29.0

### Plugin version

6.0.1

### Node.js version

16.14.2

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

12.2.1

### Description

The `openapi` option accepts an object of the type [`OpenAPIV3.Document`](https://github.com/kogosoftwarellc/open-api/blob/master/packages/openapi-types/index.ts#L271), which includes an `openapi` property to set the OpenAPI version.
`fastify-swagger` doesn't seem to pick this up, as the `openapi` version stays at the [default value of `3.0.3`](https://github.com/fastify/fastify-swagger/blob/master/lib/spec/openapi/utils.js#L45).

### Steps to Reproduce

```
await app.register(fastifySwagger, {
  routePrefix: '/documentation',
  openapi: {
    openapi: '3.1.0', // <--- This is allowed in the TypeScript interface, but is not reflected in the generated OpenAPI schema
    info: {
      title: 'Test swagger',
      description: 'Testing the Fastify swagger API',
      version: '0.1.0',
    },
  }
});
```

### Expected Behavior

_No response_

### PR DESCRIPTION
**Pull request by TimonVS**
Fixes #601

### CODE REVIEW

**Comment by TimonVS** [comment]
I’ll fix the linting error tomorrow when I’m behind a computer :)

File: lib/spec/openapi/utils.js [hunk 1]
```diff
@@ -35,14 +35,15 @@ function prepareDefaultOptions (opts) {
     transform,
     hiddenTag,
     extensions,
-    hideUntagged
+    hideUntagged,
+    ...openapi
```
Comment (climba03003) [inline lib/spec/openapi/utils.js:39]: Why spread operator at below?

File: lib/spec/openapi/utils.js [hunk 2]
```diff
@@ -35,14 +35,15 @@ function prepareDefaultOptions (opts) {
...
   }
 }
 
 function prepareOpenapiObject (opts) {
   const pkg = readPackageJson()
   const openapiObject = {
-    openapi: '3.0.3',
+    openapi: opts.openapi || '3.0.3',
```
Comment (climba03003) [inline lib/spec/openapi/utils.js:46]: Can you move to the `if` statement like below? For consistency only.

File: lib/spec/openapi/utils.js (hunk 1 above)
Comment (TimonVS) [reply lib/spec/openapi/utils.js:39]: Good point, I'm now overriding all the default `null` values for `info`, `servers`, etc. Is it okay if I use spread at the top? It diverges from the pattern that was set before me. Is there a reason `info` etc. have to be set to `null`?

File: lib/spec/openapi/utils.js [hunk 3]
```diff
@@ -35,7 +36,7 @@ function prepareDefaultOptions (opts) {
     transform,
     hiddenTag,
     extensions,
-    hideUntagged
+    hideUntagged,
```
[the remark this answers is not in the document]
Comment (TimonVS) [reply lib/spec/openapi/utils.js]: Sorry, I'm so used to Prettier autofixing my code that I missed this one 😅

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by TimonVS** [comment]
@mcollina this MR targets the `master` branch, which only supports Fastify v4. Should I open an MR to the 6.x branch (doesn't exist yet) to backport the fix for Fastify v3 support?

**Review by climba03003 — APPROVED** [review]
LGTM.

**Comment by mcollina** [comment]
> @mcollina this MR targets the `master` branch, which only supports Fastify v4. Should I open an MR to the 6.x branch (doesn't exist yet) to backport the fix for Fastify v3 support?

I've created a 6.x branch, go ahead!

**Pull request merged**